nsxt/resource_nsxt_policy_edge_high_availability_profile.go

```diff
@@ -0,0 +1,298 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/sites/enforcement_points"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+
+	"github.com/vmware/terraform-provider-nsxt/nsxt/metadata"
+)
+
+var policyEdgeHighAvailabilityProfileSchema = map[string]*metadata.ExtendedSchema{
+	"nsx_id":       metadata.GetExtendedSchema(getNsxIDSchema()),
+	"path":         metadata.GetExtendedSchema(getPathSchema()),
+	"display_name": metadata.GetExtendedSchema(getDisplayNameSchema()),
+	"description":  metadata.GetExtendedSchema(getDescriptionSchema()),
+	"revision":     metadata.GetExtendedSchema(getRevisionSchema()),
+	"tag":          metadata.GetExtendedSchema(getTagsSchema()),
+	"site_path": {
+		Schema: schema.Schema{
+			Type:         schema.TypeString,
+			Description:  "Path to the site this Host Transport Node belongs to",
+			Optional:     true,
+			ForceNew:     true,
+			Default:      defaultInfraSitePath,
+			ValidateFunc: validatePolicyPath(),
+		},
+		Metadata: metadata.Metadata{
+			SchemaType: "string",
+			Skip:       true,
+		},
+	},
+	"enforcement_point": {
+		Schema: schema.Schema{
+			Type:        schema.TypeString,
+			Description: "ID of the enforcement point this Host Transport Node belongs to",
+			Optional:    true,
+			ForceNew:    true,
+			Default:     "default",
+		},
+		Metadata: metadata.Metadata{
+			SchemaType: "string",
+			Skip:       true,
+		},
+	},
+	"bfd_probe_interval": {
+		Schema: schema.Schema{
+			Type:     schema.TypeInt,
+			Optional: true,
+			Default:  500,
+		},
```

Comment:
MP resource has more refined validations - f.e, for this attribute we have `validation.IntBetween(50, 60000)`. Policy spec also specifies those - I think we should be consistent with MP resource here

Reply:
I'm a bit confused as so far we were reluctant to validate such things in the provider, and left those to NSX.
As this resource is auto-generated, maybe we should add these validations in the generator? Those ranges are available in the APi spec...

Reply:
I was just pointing out inconsistency with MP resource, not, not insisting on the limits